src/Type/Row.purs

```diff
@@ -0,0 +1,27 @@
+module Type.Row
+  ( class RowLacks
+  ) where
+
+-- Must not be exported
+foreign import data RowLacking :: Symbol -> Symbol -> Type
+
+class RowLacksF (key :: Symbol)
+                (row :: # Type)
+                (typ :: Type)
+
+instance rowLacksF
+  :: ( RowCons key typ () singleton
+     , Union row singleton composed
+     , RowCons key typ rest composed )
+  => RowLacksF key row typ
+
+-- | Encodes the constraint that a given row does not contain a
+-- | specific key.
+class RowLacks (key :: Symbol)
+               (row :: # Type)
+
+instance rowLacks
+  :: ( RowLacksF key row (RowLacking "Existing key found in row" key)
+     , RowLacksF key row (RowLacking "Existing key found polymorphically in row" key) )
```

Comment:
It is for this case:

```purescript
test3 :: Int
test3 = 0 :: forall t. RowLacks "k" (k :: t) => Int
```

If we only have the first constraint, then `t` will unify with `(RowLacking ...)`.  But with both constraints, it can only unify with one of them.